build-now.lua

```diff
@@ -359,16 +359,16 @@ local function pos_cmp(a, b)
     return utils.compare(a.z, b.z)
 end
 
-local function create_and_link_construction(pos, item, top_of_wall)
+local function create_and_link_construction(pos, item, top_of_wall, use_open_space_for_original_tile)
     local construction = df.construction:new()
     utils.assign(construction.pos, pos)
     construction.item_type = item:getType()
     construction.item_subtype = item:getSubtype()
     construction.mat_type = item:getMaterial()
     construction.mat_index = item:getMaterialIndex()
     construction.flags.top_of_wall = top_of_wall
-    construction.flags.no_build_item = not top_of_wall
-    construction.original_tile = dfhack.maps.getTileType(pos)
+    construction.flags.no_build_item = false
```

Comment:
We should preserve `no_build_item = true` for most constructions as it improves framerate (see `cleanconst`)

Reply:
but only for "real" constructions, not top_of_wall pseudo-constructions

Reply:
Right. So should this line be unchanged?

Reply:
I think so. `no_build_item` should always be set to false by `build-now`.

Reply:
Wait, but code is for all constructions.

Reply:
Also, reverting it to construction.flags.no_build_item = not top_of_wall as it was seems to fix the problem wtih not getting build items back when deconstructing a build-now wall...